clicky-core/src/devices/display/hd66753.rs

```diff
@@ -155,15 +157,25 @@ impl Hd66753 {
             };
 
             let cgram_window = cgram
-                .chunks_exact(EMU_CGRAM_WIDTH * 2 / 8 / 2)
-                .take(height)
-                .flat_map(|row| row.iter().take(CGRAM_WIDTH * 2 / 8 / 2).rev());
+                    .chunks_exact(EMU_CGRAM_WIDTH * 2 / 8 / 2)
+                    .take(height)
+                    .flat_map(|row| {
+                        match ireg.sgs {
```

Comment:
Ehh, I think this might be a _fine_ way to factor this code.

Maybe the "best" thing to do is to replace all this iterator code with some bog-standard for loops with manual indexing... but idk if that'd be more or less readable.

I don't really mind the use of Either here, so up to you if you want to keep noodling with this code

Reply:
I'll stick with Either here.